R/t-distribution

```diff
@@ -0,0 +1,270 @@
+# App: t_distribution
+# Last update: 15 May 22
+
+library(shiny)
+library(miniUI)
+library(tidyverse)
+library(glue)
+library(Cairo)
+
+options(shiny.usecairo = TRUE)
+options(shiny.autoreload = TRUE)
+
+PRECISION <- 0.05
+EXTRA_POINTS_AROUND_CRITICAL_VALUES = c(-0.001, 0, 0.001)
+STARTING_T_VALUE <- 2.25
+STARTING_DF <- 2
+STARTING_P_VALUE <-
+  pt(STARTING_T_VALUE / 2, STARTING_DF, lower.tail = STARTING_T_VALUE <= 0) %>% round(3)
+RANGE <- c(-4, 4) # range for T value
+
+label_p_value_html <- function(t, df) {
+  glue(
+    "\\(P(T {ifelse(t <= 0, '\\\\leq', '\\\\geq')} {round(t, 2)}) = {ifelse(t <= 0, pt(t, df), 1 - pt(t, df)) %>% abs() %>% round(3)}\\)"
+  )
+}
+
+label_p_value <- function(t, df) {
+  glue(
+    "P(T {ifelse(t <= 0, '<=', '>=')} {round(t, 2)}) == {ifelse(t <= 0, pt(t, df), 1 - pt(t, df)) %>% abs() %>% round(3)}"
+  )
+}
+
+## UI ---------------------------------------------------------
+ui <- miniPage(
+  title = "t distribution",
+  theme = bslib::bs_theme(version = 4),
+  uiOutput("css"),
+  withMathJax(),
+  miniContentPanel(
+    conditionalPanel(
+      "output.help",
+      div(
+        class = "alert alert-info p-1 rounded d-flex align-items-center",
+        icon("warning", class = "h3 mx-3", verify_fa = FALSE),
+        
+        p(
+          class = "mb-0",
+          "This app has three modes: critical values, p-values, and combined. You can select the mode by adding",
+          code("?mode=critical_values"),
+          ",",
+          code("?mode=p_values"),
+          ", or",
+          code("?mode=combined"),
+          "to the url. You are currently viewing the combined mode. Choosing a mode will suppress this message."
+        )
+      )
+    ),
+    plotOutput("distribution", height = "100%")
+  ),
+  miniButtonBlock(
+    style = "justify-content: space-around; gap: 1em; padding: .5em;",
+    # critical values
+    conditionalPanel( # set significance level
+      "output.mode == 'combined' || output.mode == 'critical_values'",
+       div(
+         id = "critical-value-panel",
+         class = "d-flex flex-column",
+         
+         div(# set alpha
+          class = "d-flex",
+          sliderInput("alpha", "Type I error, \\(\\alpha\\)", 0.01, 0.2, 0.05, 0.01)
+          ),
+         div(# degree of freedom
```

Comment:
df is required in all modes, it also affects p-values, move outside of conditional panel